lib/ansible/modules/tacp/tacp_instance.py

```diff
@@ -80,278 +78,727 @@
+            necessary to populate an ApiCreateApplicationPayload.
+
+    Returns:
+        ApiCreateApplicationPayload: A populated payload for creating
+            a new application instance.
+    """
+    create_new_instance_payload = tacp.ApiCreateApplicationPayload(
+        name=parameters_to_create_new_application.get('instance_name'),
+        datacenter_uuid=parameters_to_create_new_application.get(
+            'datacenter_uuid'),
+        flash_pool_uuid=parameters_to_create_new_application.get(
+            'storage_pool_uuid'),
+        migration_zone_uuid=parameters_to_create_new_application.get(
+            'migration_zone_uuid'),
+        template_uuid=parameters_to_create_new_application.get(
+            'template_uuid'),
+        vcpus=parameters_to_create_new_application.get('vcpus'),
+        memory=parameters_to_create_new_application.get('memory'),
+        vm_mode=parameters_to_create_new_application.get('vm_mode'),
+        networks=parameters_to_create_new_application.get('networks'),
+        vnics=parameters_to_create_new_application.get('vnics'),
+        boot_order=parameters_to_create_new_application.get('boot_order'),
+        hardware_assisted_virtualization_enabled=parameters_to_create_new_application.get('vtx_enabled'),  # noqa
+        enable_automatic_recovery=parameters_to_create_new_application.get(
+            'auto_recovery_enabled'),
+        description=parameters_to_create_new_application.get('description'),
+        application_group_uuid=parameters_to_create_new_application.get(
+            'application_group_uuid')
     )
 
-    # seed the result dict in the object
-    # we primarily care about changed and state
-    # change is if this module effectively modified the target
-    # state will include any data that you want your module to pass back
-    # for consumption, for example, in a subsequent task
-    result = dict(
-        changed=False,
-        args=[]
+    return create_new_instance_payload
+
+
+def create_instance(playbook_instance):
+    """Create an application instance and return the API response object.
+
+    Args:
+        playbook_instance (dict): The specified instance configuration from
+        the playbook input
+
+    Returns:
+        ApiResponsePayload: The response from the create application request.
+    """
+    parameters_to_create_new_application = get_parameters_to_create_new_application(  # noqa
+        playbook_instance)
+    create_new_instance_payload = get_create_new_instance_payload(
+        parameters_to_create_new_application)
+
+    created_instance_response = RESOURCES['app'].create(create_new_instance_payload)  # noqa
+
+    return created_instance_response
+
+
+def instance_power_action(instance_uuid, action):
+    RESOURCES['app'].power_action_on_instance_by_uuid(
+        instance_uuid, action
     )
+    RESULT['changed'] = True
+
+
+def get_instance_by_name(instance_name):
+    instance = RESOURCES['app'].filter(name=instance_name)[0]
```

Comment:
There are a whole bunch of places in the code where you are blindly indexing into an iteratable (with [0]). This is really dangerous if someone else uses this method without checking that the instance exists. Ideal, you should be returning `None` here if the instance is not found.